src/docbuild/models/config_model/env.py

```diff
@@ -0,0 +1,268 @@
+
+
+class LanguageCode(str):
+    """Placeholder for the LanguageCode type."""
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source_type: Any, handler: Any):
+        return handler(str)
+
+
+# --- Custom Types and Utilities ---
+
+# A type for domain names, validated with a regex.
+DomainName = Annotated[
+    str,
+    Field(
+        pattern=r"^(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,63}$",
+        title="Valid Domain Name",
+        description="A string representing a fully qualified domain name (FQDN).",
+        examples=["example.com", "sub.domain.net"],
+    ),
+]
+
+
+# --- Configuration Models ---
+
+class Env_Server(BaseModel):
+    """Defines server settings."""
+
+    model_config = ConfigDict(extra='forbid')
+
+    name: str = Field(
+        title="Server Name",
+        description="A human-readable identifier for the environment/server.",
+        examples=["Production-EU", "Dev-Local"],
+    )
+    "The descriptive name of the server."
+
+    role: Literal["development", "production", "staging"] = Field(
+        title="Server Role",
+        description="The operational role of the environment.",
+        examples=["production"],
+    )
+    "The environment type, used for build behavior differences."
+
+    host: IPvAnyAddress | DomainName = Field(
+        title="Server Host",
+        description="The hostname or IP address the documentation is served from.",
+        examples=["127.0.0.1", "docserver.example.com"],
+    )
+    "The host address for the server."
+
+    port: int | None = Field(
+        None,
+        title="Server Port",
+        description="The port number on which the server is listening.",
+        examples=[8080, 443],
+    )
+    "The port used by the server, or None if inferred."
+
+    enable_mail: bool = Field(
+        title="Enable Email",
+        description="Flag to enable email sending features (e.g., build notifications).",
+        examples=[True],
+    )
+    "Whether email functionality should be active."
+
+
+class Env_GeneralConfig(BaseModel):
+    """Defines general configuration."""
+
+    model_config = ConfigDict(extra='forbid')
+
+    default_lang: LanguageCode = Field( 
+        title="Default Language",
+        description="The primary language code (e.g., 'en') used for non-localized content.",
+        examples=["en"],
+    )
+    "The default language code."
+
+    languages: list[LanguageCode] = Field( 
+        title="Supported Languages",
+        description="A list of all language codes supported by this documentation instance.",
+        examples=[["en", "de", "fr"]],
+    )
+    "A list of supported language codes."
+
+    canonical_url_domain: HttpUrl = Field(
+        title="Canonical URL Domain",
+        description="The base domain used to construct canonical URLs for SEO purposes.",
+        examples=["https://docs.example.com"],
+    )
+    "The canonical domain for URLs."
+
+
+class Env_TmpPaths(BaseModel):
```

Comment:
After looking over the names, I think I've introduced some, aehm, inconsistencies.For example, we have `tmp_base_path` (suffix "path") and `tmp_build_dir` (suffix "dir").

I think it would be a good timing to fix that as well. Sorry for the inconvenience! I think there are a few locations where these names are used.

Additionally, you could also adapt `etc/docbuild/env.example.toml` too as this acts as a template.